coding/python_style_guide.rst

```diff
@@ -2,658 +2,697 @@
+Creating a more specific identifier is suggested to avoid collisions.
+In the case of *filter*, ``filterName`` may be appropriate; for *filter objects*, something like ``filterObj`` might be appropriate.
 
-No:
+.. _style-guide-py-naming-attributes:
 
-.. code-block:: py
+Class Attribute Names SHOULD be camelCase with leading lowercase
+----------------------------------------------------------------
 
-   if len(seq):
-       pass
+`Opposes PEP 8 <https://www.python.org/dev/peps/pep-0008/#id45>`__.
+Error codes: N802 and N803.
 
-   if not len(seq):
-       pass
+.. _style-guide-py-naming-functions:
 
-.. _style-guide-py-6-3:
+Module methods (free functions) SHOULD be camelCase with leading lowercase
+--------------------------------------------------------------------------
 
-A conditional test of a boolean value SHOULD not explicitly test against ``True`` or ``False``
-----------------------------------------------------------------------------------------------
+`Opposes PEP 8 <https://www.python.org/dev/peps/pep-0008/#id45>`__.
+Error code: N802.
 
-Don't compare boolean values to ``True`` or ``False`` using ``==`` (unless it matters, e.g. for tri-state logic).
+.. _style-guide-py-naming-class-modules:
 
-Yes:
+Modules which contain class definitions SHOULD be named after the class name
+----------------------------------------------------------------------------
 
-.. code-block:: py
+Modules which contain class definitions should be named after the class name (one module per class).
 
-   if greeting:
+.. _style-guide-py-naming-ext-modules:
 
-No:
+When a Python module wraps a C/C++ extension module, the C/C++ module SHOULD be named <module>Lib
+-------------------------------------------------------------------------------------------------
 
-.. code-block:: py
+When an extension module written in C or C++ has an accompanying Python module that provides a higher level (e.g. more object oriented) interface, the C/C++ module should append ``Lib`` to the module's name (e.g. ``socketLib``).
 
-   if greeting == True:
-       pass
+.. _style-guide-py-naming-ambiguous:
 
-   if greeting is True:
-       pass
+Names l (lowercase: el), O (uppercase: oh), I (uppercase: eye) MUST be avoided
+------------------------------------------------------------------------------
 
-.. _style-guide-py-pitfalls:
+Never use these characters as single character variable names:
 
-7. Programming Pitfalls
-=======================
+- ``l`` (lowercase letter el),
+- ``O`` (uppercase letter oh), or
+- ``I`` (uppercase letter eye).
 
-.. _style-guide-py-7-1:
+In some fonts, these characters are indistinguishable from the numerals one and zero.
+When tempted to use ``l``, use ``L`` instead.
 
-``if x`` SHOULD NOT be used when you mean ``if x != None``
-----------------------------------------------------------
+.. _style-guide-py-files:
 
-Beware of writing ``if x`` when you mean ``if x != None``.
```

Comment:
Did this disappear because PEP-8 says the same thing? If so, DM-6720 should be annotated to indicate that the work will be subsumed into this ticket.